src/ray/raylet/node_manager.cc

```diff
@@ -1897,9 +1918,9 @@ void NodeManager::FinishAssignedTask(Worker &worker) {
 
   // Unset the worker's assigned task.
   worker.AssignTaskId(TaskID::Nil());
+  num_scheduled_[spec.GetSchedulingClass()] -= 1;
```

Comment:
TODO: might be missing some places where this needs to be decremented / this should be handled in a nicer way

Reply:
@stephanie-wang any suggestions?